### PR TITLE
Formtypes should be placed in Form\Type-Directory

### PR DESCRIPTION
**Pull request by HKandulla**

### CODE REVIEW

**Comment by zatikbalazs** [comment]
Yes, according to the Symfony Book you should put them in the ``\Form\Type`` directory.

**Comment by xabbuh** [comment]
:+1: Though we have two `use` statements on the same page. Can you please update them too?

**Comment by HKandulla** [comment]
@xabbuh: I am not sure what two use statements you mean. Can you give me the lines?

**Comment by xabbuh** [comment]
@HKandulla Sorry, I missed the fact that you already tackled them.

**Comment by xabbuh** [comment]
@HKandulla Can you also add the filename comments like started in #5727?

**Comment by HKandulla** [comment]
@xabbuh: Good idea. I added the filename comments to all files (Formtype, Controller and Views). Additionally I changed the namespace for the example Controller from "AppBundle\Controller\Admin" to "AppBundle\Controller" to use standard directory order.

File: best_practices/forms.rst [hunk 1]
```diff
@@ -51,7 +52,8 @@ form in its own PHP class::
 
 To use the class, use ``createForm`` and instantiate the new class::
 
-    use AppBundle\Form\PostType;
+    // src/AppBundle/Controller/PostController.php
```
Comment (xabbuh) [inline best_practices/forms.rst]: I don't think we need this here (it does not really matter how your controller is named and we actually omit most of the class already for readability). Though I would now move the `use` statement below the placeholder comment.

File: best_practices/forms.rst [hunk 2]
```diff
@@ -91,6 +93,7 @@ directly in your form class, this would effectively limit the scope of that form
 
 .. code-block:: php
 
+    // src/AppBundle/Form/Type/PostType.php
```
Comment (xabbuh) [inline best_practices/forms.rst]: We should add the `namespace` declaration and after that add a placeholder comment or add a `use` statement for the `AbstractType` class.

File: best_practices/forms.rst [hunk 3]
```diff
@@ -176,6 +182,7 @@ Handling a form submit usually follows a similar template:
 
 .. code-block:: php
 
+    // src/AppBundle/Controller/PostController.php
```
Comment (xabbuh) [inline best_practices/forms.rst]: I would remove it here too (for the same reasons as above).

**Comment by WouterJ** [comment]
I'm not sure I'm 100% behind this PR. A major goal of the best practices guide is to make everything simpler and quicker to write. That's why templates have been moved to `/app/Resources` (`AppBundle:static:home.html.twig` vs `static/home.html.twig`).

For the same reason, I would argue that the `Type` prefix of the form type classes already makes it obvious that these are types. Adding a deeper level of namespaces, `Type`, means we only add more characters to write.

On the other side, things can get out of hand pretty quickly when having a form type, extension and listener. It's also consistent with the internal Form component extension organization.

So I'm leaning a bit towards -0.4 for using `Form\Type` in the best practices (but instead, updating to the book to use `Form` directly). I would also like to see the opinions of @weaverryan @javiereguiluz before merging/not-merging this.

Anyway, as always a huge thanks to you for putting time into making the Symfony documentation more consistent! It's great you noticed this inconsistency, it's just a matter of deciding the practice.

**Comment by javiereguiluz** [comment]
@WouterJ here it is the discussion I had with @weaverryan about this issue: https://github.com/symfony/symfony-demo/issues/6

**Comment by HKandulla** [comment]
Personally, I like to keep FormTypes in an own directory, especially when you start to put more/all busines logik into one Bundle. But I leave it up to you. Let me know, than I can do the changes xabbuh pointed out before you merge.

**Comment by weaverryan** [comment]
The Symfony best practices are to put them in the `Form` directory alone. As @HKandulla hinted, this is totally subjective. I dislike the idea of making a beginner create this `AppBundle/Form/Type/*` structure - where `Form` will have nothing in it (other than a `Type` directory) for many projects. For the Symfony Demo - which is maturing into a larger project - having `Form\Type` makes more sense, but you could still put all the "types" in `Form` and other things in sub-directories (`Form/DataTransformer`).

My vote is to keep it in `Form`.

**Comment by WouterJ** [comment]
As there seems to be a general vote to not place form types in `Form\Type` *by default*, I'm going to close this PR. You can of course still comment if you disagree with this decision.

In order to make things more clear and explicitly tell people that this is not a hard rule (nothing in the docs is btw), I've created a new PR adding a best practice box for the form type namespace: https://github.com/symfony/symfony-docs/pull/6059

:christmas_tree:

**Pull request closed without merging**